lib/src/cache/request_cache_key.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:dio/dio.dart';
+
+/// [RequestCacheKey] is a class that represents the key of a request in the cache.
+class RequestCacheKey {
+  const RequestCacheKey._({
+    required this.path,
+    required this.method,
+    this.cacheExpirationDate,
+  });
+
+  factory RequestCacheKey.fromRequestOptions(RequestOptions options) =>
+      RequestCacheKey._(
+        path: options.path,
+        method: options.method,
+        cacheExpirationDate: options.extra['cacheExpiresIn'] != null
+            ? DateTime.now().add(options.extra['cacheExpiresIn'])
+            : null,
+      );
+
+  factory RequestCacheKey.fromJson(Map<String, dynamic> json) {
+    return RequestCacheKey._(
+      path: json['path'] as String,
+      method: json['method'] as String,
+      cacheExpirationDate: json['cacheExpirationDate'] != null
+          ? DateTime.parse(json['cacheExpirationDate'] as String)
+          : null,
```

Comment:
Com o tryParse não funciona? 
```suggestion
      cacheExpirationDate:  DateTime.tryParse(json['cacheExpirationDate'] as String),
```